lib/_stream_readable.js

```diff
@@ -302,6 +302,8 @@ function readableAddChunk(stream, chunk, encoding, addToFront) {
       }
     } else if (!addToFront) {
       state.reading = false;
+      if (state.needReadable)
+        emitReadable(stream);
```

Comment:
This potentially causes a type of live lock if the user keeps pushing empty chunks. That's a bug from the user's side but we might want to handle that. Not sure how to avoid that. I was thinking of adding another state i.e. "emptyPush" and only emit readable for one empty push in a row... though again... unsure if required or even a good idea